headerfs/js.go

```diff
@@ -0,0 +1,402 @@
+//go:build js && wasm
+
+package headerfs
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall/js"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/walletdb"
+	"github.com/linden/indexeddb"
+)
+
+const headersStore = "headers"
+
+// headerStore combines a IndexedDB store of headers within a flat file in addition
+// to a database which indexes that flat file. Together, these two abstractions
+// can be used in order to build an indexed header store for any type of
+// "header" as it deals only with raw bytes, and leaves it to a higher layer to
+// interpret those raw bytes accordingly.
+//
+// TODO(roasbeef): quickcheck coverage.
+type headerStore struct {
+	mtx sync.RWMutex // nolint:structcheck // false positive because used as embedded struct only
+
+	idb *indexeddb.DB
```

Comment:
What if we had this package implement the `walletdb` interface? Then we could just drop it in here, as well as some other locations/projects that use that central KV-store abstraction.